app/components/c-panel.vue

```diff
@@ -0,0 +1,408 @@
+        type: [String, Number],
+        default: 0,
+        validator(value) {
+          return [0, 1, 2].includes(parseInt(value, 10));
+        },
+      },
+
+      /**
+       * Defines the border color of the panel
+       *
+       * Valid values: `yellow`, `blue`, `gray`
+       */
+      color: {
+        type: String,
+        default: 'blue',
+        validator(value) {
+          return [
+            'yellow',
+            'blue',
+            'gray'
+          ].includes(value);
+        },
+      },
+
+      /**
+       * Defines if the triangle shaped button shall be displayed or not
+       */
+      triangleButton: {
+        type: Boolean,
+        default: false
+      },
+
+      /**
+       * Defines if the triangle shaped button is initially open or not
+       */
+      triangleButtonOpen: {
+        type: Boolean,
+        default: false
+      },
+
+      /**
+       * Defines the size of the triangle button
+       *
+       * Valid values: `0`, `500`
+       */
+      triangleButtonSize: {
+        type: [Number, String],
+        default: 0,
+        validator(value) {
+          return [0, 500].includes(parseInt(value, 10));
+        },
+      },
+
+      /**
+       * Defines the plus color of the plus for the triangle button
+       *
+       * Valid values: `white`, `blue`
+       */
+      triangleButtonPlusColor: {
+        type: String,
+        default: 'white',
+        validator(value) {
+          return [
+            'white',
+            'blue'
+          ].includes(value);
+        },
+      },
+
+    },
+    data() {
+      return {
+        hasTouch: null,
+      };
+    },
+
+    computed: {
+      panelModifiers() {
+        return {
+          border: this.border.toString(), /* TODO - remove .toString() once vue-bem-cn accepts numbers */
+          color: this.color,
+        };
+      },
+      headingModifiers() {
+        return {
+          spacing: this.headingSpacing.toString(), /* TODO - remove .toString() once vue-bem-cn accepts numbers */
+        };
+      },
+      contentModifiers() {
+        return {
+          spacing: this.contentSpacing.toString(), /* TODO - remove .toString() once vue-bem-cn accepts numbers */
+        };
+      },
+      triangleButtonModifiers() {
+        return {
+          size: this.triangleButtonSize.toString(), /* TODO - remove .toString() once vue-bem-cn accepts numbers */
+        };
+      },
+      shadowModifiers() {
+        return {
+          color: this.color,
+        };
+      },
+      plusModifiers() {
+        return {
+          size: this.triangleButtonSize.toString(), /* TODO - remove .toString() once vue-bem-cn accepts numbers */
+          color: this.triangleButtonPlusColor,
+          open: this.triangleButtonOpen,
+        };
+      },
+    },
+    // watch: {},
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    mounted() {
+      this.hasTouch = 'ontouchstart' in document.documentElement ||
```

Comment:
Why did you not define this directly in the `data()` method?

Reply:
Not really - done.